libs/api/domains/passport/src/lib/passport.service.ts

```diff
@@ -0,0 +1,139 @@
+import { Inject, Injectable } from '@nestjs/common'
+import { ApolloError } from 'apollo-server-express'
+import { FetchError } from '@island.is/clients/middlewares'
+import isBefore from 'date-fns/isBefore'
+import differenceInMonths from 'date-fns/differenceInMonths'
+import {
+  IdentityDocumentApi,
+  IdentityDocumentResponse,
+} from '@island.is/clients/passports'
+import { AuthMiddleware } from '@island.is/auth-nest-tools'
+import type { Auth, User } from '@island.is/auth-nest-tools'
+import type { Logger } from '@island.is/logging'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import {
+  Gender,
+  IdentityDocumentModel,
+} from './models/identityDocumentModel.model'
+import { IdentityDocumentModelChild } from './models/identityDocumentModelChild.model'
+
+export type ExpiryStatus = 'EXPIRED' | 'LOST'
+const LOG_CATEGORY = 'passport-service'
+
+@Injectable()
+export class PassportService {
+  constructor(
+    @Inject(LOGGER_PROVIDER)
+    private logger: Logger,
+    private passportsApi: IdentityDocumentApi,
+  ) {}
+
+  handleError(error: any, detail?: string): ApolloError | null {
+    this.logger.error(detail || 'Domain passport error', {
+      error: JSON.stringify(error),
+      category: LOG_CATEGORY,
+    })
+    throw new ApolloError('Failed to resolve request', error.status)
+  }
+
+  private handle4xx(error: FetchError) {
+    if (error.status === 403 || error.status === 404) {
+      return undefined
+    }
+    this.handleError(error)
+  }
```

Comment:
I had initially (few months ago) assumed the service would return a 404 when no passport was available. Now that the service is ready, that's not the case, it will return an empty array.